conans/test/functional/autotools_configure_test.py

```diff
@@ -43,13 +43,95 @@ def test_mocked_methods(self):
         things = "'things'" if platform.system() != "Windows" else "things"
         self.assertEquals(runner.command_called, "make %s -j%s" % (things, cpu_count()))
 
+    def test_cppstd(self):
+        # Valid one for GCC
+        settings = MockSettings({"build_type": "Release",
+                                 "arch": "x86",
+                                 "compiler": "gcc",
+                                 "compiler.libcxx": "libstdc++11",
+                                 "compiler.version": "7.1"})
+        options = MockOptions({"cppstd": "17"})
+        conanfile = MockConanfile(settings, options)
+        be = AutoToolsBuildEnvironment(conanfile)
+        expected = be.vars["CXXFLAGS"]
+        self.assertIn("-std=c++1z", expected)
+
+        # Invalid one for GCC
+        settings = MockSettings({"build_type": "Release",
+                                 "arch": "x86",
+                                 "compiler": "gcc",
+                                 "compiler.libcxx": "libstdc++11",
+                                 "compiler.version": "4.9"})
+        options = MockOptions({"cppstd": "17"})
```

Comment:
I am wondering if this should be managed somehow. So you requested cppstd=17, which is impossible, it will be ignored, but it will affect the binary package ID? Maybe should it raise?

Reply:
Probably shouldn't be a valid option if you used the "add_cppstd" and will raise because of it. (not in this case of Mock()). 
Actually we miss information in the case of Visual Studio, where depending on the revision of the compiler the default standard can change. In visual 2015 rev3 the default is cpp14, and c++11 makes no sense, there is no flag to specify a lower one, but in previous revisions the default was c++11.